main.py

```diff
@@ -173,7 +210,7 @@ def code_with_after(tree, after):
     elif type(tree) is ast.Expression:
         return code(tree.body)
     elif type(tree) is ast.ExtSlice:
-        return ', '.join([code(dim) for dim in tree.dims])
+        return ' '.join(code(dim) + ',' for dim in tree.dims)
```

Comment:
One of the things I discovered while writing the slice_torture test is that objects are allowed to treat the extended slices `a[i,]`, `a[i:j:k,]`, `a[...,]` differently from the plain slices `a[i]`, `a[i:j:k]`, `a[...]`. The trailing comma makes sure there’s still a tuple when there’s only one entry. It could be elided when `len(tree.dims) > 1`, if you like.